server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -4198,20 +4201,36 @@ private PhysicalNetworkServiceProvider addDefaultBaremetalProvidersToPhysicalNet
             addProviderToPhysicalNetwork(physicalNetworkId, "BaremetalUserdataProvider", null, null);
         } else if (dvo.getNetworkType() == NetworkType.Advanced) {
             addProviderToPhysicalNetwork(physicalNetworkId, "BaremetalPxeProvider", null, null);
-            enableBaremetalProvider("BaremetalPxeProvider");
+            enableProvider("BaremetalPxeProvider");
         }
 
         return null;
     }
 
-    private void enableBaremetalProvider(String providerName) {
+    private void enableProvider(String providerName) {
         QueryBuilder<PhysicalNetworkServiceProviderVO> q = QueryBuilder.create(PhysicalNetworkServiceProviderVO.class);
         q.and(q.entity().getProviderName(), SearchCriteria.Op.EQ, providerName);
         PhysicalNetworkServiceProviderVO provider = q.find();
         provider.setState(PhysicalNetworkServiceProvider.State.Enabled);
         _pNSPDao.update(provider.getId(), provider);
     }
 
+    private PhysicalNetworkServiceProvider addConfigDriveToPhysicalNetwork(long physicalNetworkId) {
+        PhysicalNetworkVO pvo = _physicalNetworkDao.findById(physicalNetworkId);
+        DataCenterVO dvo = _dcDao.findById(pvo.getDataCenterId());
+        if (dvo.getNetworkType() == NetworkType.Advanced) {
```

Comment:
Why wouldn't config-drive be supported in Basic Networking?

It's a network provider, correct?

Reply:
No technical reason but there is very limited knowledge about Basic networking within Nuage team; it would be best that other community members further extend it. We could leave out the check but it might not just work like that.

Reply:
Understood! Thanks, I'll test that when I get the chance, but that won't be very soon :(

Don't wait with merging before my tests

Reply:
@wido you suggest we remove the check ?

Reply:
@krissterckx leave it in, for @wido to remove hen he has test-time.

Reply:
Indeed @DaanHoogland and @krissterckx leave the check in. Once I have the time to fully test it and verify it also works in Basic Networking I'll submit a PR for that.

Reply:
Got it and yes, agree. +1